packages/flutter_tools/lib/src/ios/devices.dart

```diff
@@ -287,13 +291,22 @@ class IOSDevice extends Device {
         await diagnoseXcodeBuildFailure(buildResult);
         printError('');
         return LaunchResult.failed();
+      }else{
+        final String plistPath = fs.path.join(buildResult.output, 'Info.plist');
```

Comment:
Can you get PRODUCT_BUNDLE_IDENTIFIER from buildResult.xcodeBuildExecution.buildSettings instead of parsing the plist?

Reply:
Actually, I tried that, but it doesn't work, at least not on the project that I'm working on. I'm not sure why. 
when I change the code to
```
 final packageId1 = buildResult.xcodeBuildExecution.buildSettings["PRODUCT_BUNDLE_IDENTIFIER"];
        final String plistPath = fs.path.join(buildResult.output, 'Info.plist');
        final packageId2 = PlistParser.instance.getValueFromFile(
          plistPath,
          PlistParser.kCFBundleIdentifierKey,
        );
        printError('from plist: $packageId1 from buildSettings: $packageId2 equals? ${packageId1 == packageId2}');
```
und build with `flavor --dev`
the output is:
`from plist: com.diagnosia.fachinformation from buildSettings: com.diagnosia.fachinformation.dev equals? false`
It seems that the PRODUCT_BUNDLE_IDENTIFIER is also set to the default value.

Also, parsing Info.plist is consistent with `startApp()` from `simulators.dart`, I think it makes sense to use the same approach in both cases to get consistent behavior.

Reply:
[This user](https://github.com/flutter/flutter/issues/21335#issuecomment-423595855) also describes the same behavior of PRODUCT_BUNDLE_IDENTIFIER that i noticed.

Reply:
PRODUCT_BUNDLE_IDENTIFIER should match up with CFBundleIdentifier in the Info.plist.  It sounds like that's the real problem.
CFBundleIdentifier in the Info.plist should become $(PRODUCT_BUNDLE_IDENTIFIER), and PRODUCT_BUNDLE_IDENTIFIER should be what's populated by the template.

Reply:
indeed, this seems to fix the problem as of now.
I followed the instructions linked at https://flutter.dev/docs/deployment/flavors
which seems to be incomplete

Reply:
Let's fall back to PRODUCT_BUNDLE_IDENTIFIER in buildResult.xcodeBuildExecution.buildSettings instead of parsing the plist.  It sounds like this might partly be a documentation issue.  Changing the bundle identifiers per build configuration should be done in the build settings, not in the Info.plist, which is already set to $(PRODUCT_BUNDLE_IDENTIFIER).

![Screen Shot 2019-10-23 at 2 26 16 PM](https://user-images.githubusercontent.com/682784/67435575-350b8d00-f5a1-11e9-917a-fb82b73fad15.png)

Reply:
Yes, to me it seems that this would fix the issue, given that the $(PRODUCT_BUNDLE_IDENTIFIER) and CFBundleIdentifier are correctly configured. It's weird that Xcode doesn't complain if there is a mismatch.

Better documentation on how to setup flavors correctly would be great, I agree.

I will update my PR tomorrow to use buildResult.xcodeBuildExecution.buildSettings.